### PR TITLE
Bayesian Model Averaged T-Test

### PR DESCRIPTION
**Pull request by FBartos**
Moved from: https://github.com/jasp-stats/jaspTTests/pull/205

Adds Bayesian Model Averaged T-Test analysis to JASP.

Currently requires (both merged on CRAN already): 
RoBTT: `FBartos/RoBTT@JASP-updates2`
BayesTools: `FBartos/BayesTools@JASP-RoBTT-fixes`

### CODE REVIEW

File: R/robttbayesianmodelaveragedinternal.R [hunk 1]
```diff
@@ -0,0 +1,1242 @@
...
+        tmpPrior <- try(.robttExtractPriorsFromOptions(options[[priorElements[i]]][[1]]))
+        if (jaspBase::isTryError(tmpPrior))
+          .quitAnalysis(tmpPrior)
+      }
+
+      object[[priorElements[i]]] <- tmpPrior
+    }
+  }
+
+  saveRDS(object, file = "C:/JASP/priors.RDS")
```
Comment (vandenman) [inline R/robttbayesianmodelaveragedinternal.R:204]:
```suggestion
```
:wink:

File: R/robttbayesianmodelaveragedinternal.R [hunk 2]
```diff
@@ -0,0 +1,1241 @@
...
+    warmup  = options[["advancedMcmcAdaptation"]],
+    iter    = options[["advancedMcmcAdaptation"]] + options[["advancedMcmcSamples"]],
+    thin    = options[["advancedMcmcThin"]],
+    # additional settings
+    convergence_checks = RoBTT::set_convergence_checks(
+      max_Rhat            = 1.05,
+      min_ESS             = 500
+    ),
+    save     = "all",
+    seed     = options[["seed"]], # TODO: .getSeedJASP(options)
```
Comment (vandenman) [inline R/robttbayesianmodelaveragedinternal.R:575]: todo?

**Comment by vandenman** [comment]
The model with `contExpon` and `contBinom` takes _forever_ to run the last models:
image

Am I doing something wrong or missing something?

**Review by vandenman — CHANGES_REQUESTED** [review]
The code looks fine, other than the one bug I think everything is good to merge!

File: R/robttbayesianmodelaveragedinternal.R (hunk 2 above)
Comment (FBartos) [reply R/robttbayesianmodelaveragedinternal.R:575]: thanks, fixed!
(the option didn't work because of the R-version logic operations change bug)

**Comment by FBartos** [comment]
also fixed the prior plot bug now

When installing the module on a nightly, I'm getting this message
image
(I can just close it an it works but it's a bit odd)

**Comment by FBartos** [comment]
The error message resolves on the new R4.31 nightly

**Review by vandenman — APPROVED** [review]

**Pull request merged**